control-plane/csi-driver/src/bin/node/mount.rs

```diff
@@ -351,3 +353,22 @@ async fn wait_file_removal(
         )),
     }
 }
+
+/// If the filesystem uuid doesn't match with the provided uuid, unmount the device.
+pub(crate) fn unmount_on_fs_id_diff(
+    device_path: &str,
+    fs_staging_path: &str,
+    volume_uuid: &Uuid,
+) -> Result<(), String> {
+    let probed_uuid = FileSystem::get_property(device_path, "UUID").map_err(|error| {
```

Comment:
should we catch this error and umount it?

Reply:
You mean unmount on both error and fs id diff?

Reply:
Yes